packages/theme/src/cli/commands/theme/check.ts

```diff
@@ -1,15 +1,94 @@
+import {themeFlags} from '../../flags.js'
 import Command from '@shopify/cli-kit/node/base-command'
 import {execCLI2} from '@shopify/cli-kit/node/ruby'
-import {cli} from '@shopify/cli-kit'
+import {cli, path} from '@shopify/cli-kit'
+import {Flags} from '@oclif/core'
 
 export default class Check extends Command {
   static description = 'Validate the theme'
 
   static flags = {
     ...cli.globalFlags,
+    ...themeFlags,
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    'auto-correct': Flags.boolean({
+      char: 'a',
+      required: false,
+      description: 'Automatically fix offenses',
+      env: 'SHOPIFY_FLAG_AUTO_CORRECT',
+    }),
+    category: Flags.string({
+      char: 'c',
+      required: false,
+      description: `Only run this category of checks
+Runs checks matching all categories when specified more than once`,
+      env: 'SHOPIFY_FLAG_CATEGORY',
+    }),
+    config: Flags.string({
+      char: 'C',
+      required: false,
+      description: `Use the config provided, overriding .theme-check.yml if present
+Use :theme_app_extension to use default checks for theme app extensions`,
+      env: 'SHOPIFY_FLAG_CONFIG',
+    }),
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    'exclude-category': Flags.string({
+      char: 'x',
+      required: false,
+      description: `Exclude this category of checks
+Excludes checks matching any category when specified more than once`,
+      env: 'SHOPIFY_FLAG_EXCLUDE_CATEGORY',
+    }),
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    'fail-level': Flags.string({
+      required: false,
+      description: 'Minimum severity for exit with error code',
+      env: 'SHOPIFY_FLAG_FAIL_LEVEL',
+      options: ['error', 'suggestion', 'style'],
+    }),
+    init: Flags.boolean({
+      required: false,
+      description: 'Generate a .theme-check.yml file',
+      env: 'SHOPIFY_FLAG_INIT',
+    }),
+    list: Flags.boolean({
+      required: false,
+      description: 'List enabled checks',
+      env: 'SHOPIFY_FLAG_LIST',
+    }),
+    output: Flags.string({
+      char: 'o',
+      required: false,
+      description: 'The output format to use',
+      env: 'SHOPIFY_FLAG_OUTPUT',
+      options: ['text', 'json'],
+      default: 'text',
+    }),
+    print: Flags.boolean({
+      required: false,
+      description: 'Output active config to STDOUT',
+      env: 'SHOPIFY_FLAG_PRINT',
+    }),
+    version: Flags.boolean({
+      char: 'v',
+      required: false,
+      description: 'Print Theme Check version',
+      env: 'SHOPIFY_FLAG_VERSION',
+    }),
   }
 
   async run(): Promise<void> {
-    await execCLI2(['theme', 'check'])
+    const {flags} = await this.parse(Check)
+    const passThroughFlags: string[] = []
+    for (const [label, value] of Object.entries(flags)) {
```

Comment:
Given the long list of flags, I elected to try a new pattern here. Rather than explicitly declaring each flag (as we do [here](https://github.com/Shopify/cli/blob/main/packages/theme/src/cli/commands/theme/open.ts#L39-L49) and could break each time we add a new flag), I created a generalizable system of choosing which flags to pass through and how. What do we think?

Reply:
I like it, as long as the flags have the same name this is a cool generic way of doing it.
Does this work if the user uses chars instead of the full flag? or at this point we always receive the full flag?